synapseclient/__main__.py

```diff
@@ -111,10 +112,25 @@ def query(args, syn):
 
 def _getIdsFromQuery(queryString, syn):
     """Helper function that extracts the ids out of returned query."""
+
+    queryType = 'synapse'
+
     ids = []
-    for item in  syn.chunkedQuery(queryString):
-        key = [k for k in  item.keys() if k.split('.', 1)[1]=='id'][0]
-        ids.append(item[key])
+
+    if re.search('from syn\d', queryString.lower()):
+
+        tbl = syn.tableQuery(queryString)
+
+        check_for_id_col = filter(lambda x: x.get('id'), tbl.headers)
+        assert check_for_id_col, ValueError("Query does not include the id column.")
+
+        # Filed issue for below (adding syn explicitly to id) - PLFM-4207
+        # This code should be removed once that is fixed.
+        ids = ['syn%s' % (x['id'], ) for x in csv.DictReader(file(tbl.filepath))]
+    else:
```

Comment:
Since we have the abililty to handle versioned entities here, should we?  I.e. by adding version information we could fetch the specific versions of entities and handle "data freezes"